File: week4discussionOOPCompare/java/Fruit.java
public class Fruit {
    private String name;
    private String color;

    private String toLowerCase(String input) {
        return input.toLowerCase();
    }

    private String capitalizeFirstLetter(String input) {
        return input.substring(0, 1).toUpperCase() + input.substring(1);
    }

    public void setName(String name) {
        this.name = toLowerCase(name);
    }

    public String getName() {
        return capitalizeFirstLetter(name);
    }

    public void setColor(String color) {
        this.color = toLowerCase(color);
    }

    public String getColor() {
        return capitalizeFirstLetter(color);
    }
}

File: week4discussionOOPCompare/javascript/fruit.js
class Fruit {
    constructor() {
      this._name = '';
      this._color = '';
    }
  
    _toLowerCase(input) {
      return input.toLowerCase();
    }
  
    _capitalizeFirstLetter(input) {
      return input.charAt(0).toUpperCase() + input.slice(1);
    }
  
    set name(name) {
      this._name = this._toLowerCase(name);
    }
  
    get name() {
      return this._capitalizeFirstLetter(this._name);
    }
  
    set color(color) {
      this._color = this._toLowerCase(color);
    }
  
    get color() {
      return this._capitalizeFirstLetter(this._color);
    }
  }

  module.exports = { Fruit };
  